pylint/checkers/logging.py

```diff
@@ -256,11 +257,18 @@ def _check_log_method(self, node: nodes.Call, name: str) -> None:
                     node=node,
                     args=(self._helper_string(node),),
                 )
-        elif isinstance(node.args[format_pos], nodes.Call):
-            self._check_call_func(node.args[format_pos])
-        elif isinstance(node.args[format_pos], nodes.Const):
+        elif isinstance(format_arg, nodes.Call):
+            self._check_call_func(format_arg)
+        elif isinstance(format_arg, nodes.Const):
             self._check_format_string(node, format_pos)
-        elif isinstance(node.args[format_pos], nodes.JoinedStr):
+        elif isinstance(format_arg, nodes.JoinedStr):
+            if any(
+                "%s" in val.value
```

Comment:
There's a lot of potential other formatting here %r %d, %0.3f, %10s... Maybe this isn't the right approach.

Reply:
Yeah that's what I assumed I'd hear back. I figured '%s' is the most common, maybe d, r, and we could at least get rid of that FP. Any better way to detect %x formatting?

Reply:
I'm not sure but I think it's not easy to solve. @DanielNoord you worked a lot on f-string could you confirm ?

Reply:
Yeah this seems to open a can of worms, although the same could be said for the original issue.

I don't know, we might just try this and see if other requests pop up.

Reply:
Unless we think the perf hit is egregious (which I don't since this is in a very specific code section), having this code or maybe I could refactor to a well named method `_check_str_formatting` or something, and can add a few more formatting types like %d or %r which should cover probably 95% of what we'd ever see in this pattern.

Reply:
+1 for checking only the most common one. The primer already show that this is useful. Let's do %s, %d, %f and %r. Probably something like is ``'%' in val.value and any(x in val.value for x in [%s, %d, %f, %r])``. Not sure if it's the best performance wise but probably better than ``any(x in val.value for x in [%s, %d, %f, %r])``